docs/conf.py

```diff
@@ -16,7 +16,20 @@
 # import sys
 # sys.path.insert(0, os.path.abspath('.'))
 
-from clickhouse_sqlalchemy import __version__, VERSION
+from pathlib import Path
+from sys import version_info
+
+PY_VERSION = version_info[:3]
+if PY_VERSION >= (3, 11, 0):
+    import tomllib as tomlib
+else:
+    import tomli as tomlib
+
+PYPROJECT = Path(__file__).parent.parent / "pyproject.toml"
+with open(PYPROJECT, mode="rb") as ppt:
+    data = tomlib.load(ppt)
+    _version: str = data["tool"]["poetry"]["version"]
+    VERSION = tuple(map(int, _version.split(".")))
```

Comment:
@DaniilAnichin , @xzkostyan , I decided to create docs group and parse the `pyproject.toml` dynamically when the `docs/conf.py` script is in effect. I am not sure this solution is optimal, but I was in pursuit of two things:

1. the `pyproject.toml` file should be the sole metadata source
2. additional poetry plugins or backend wrappers to be avoided

Please judge this snippet severely.

Reply:
@DaniilAnichin , thank you for your feedback. I found this [package](https://pypi.org/project/poetry-dynamic-versioning/) enabling dynamic versioning, but I am not sure it is a good idea to mess with poetry plug-ins.

The main idea is in enhancing the project with `pyproject.toml` and then choosing a back-end, maybe [setuptools](https://pypi.org/project/setuptools/) will fit more or [hatch](https://hatch.pypa.io/dev/) will do enough or [pdm](https://pdm-project.org/latest/) will make things easier.

About poetry plug-ins: https://python-poetry.org/docs/plugins/#using-plugins .

UPD: maybe pdm will do like in fastapi... https://github.com/fastapi/fastapi/blob/master/pyproject.toml#L7 . I'll make another PR to compare the results.

Reply:
@Polyrom, could you have a look at his part? You may find more elegant solution for what joins the project and sphinx via dynamic versioning (or maybe without it).